hibernate-core/src/main/java/org/hibernate/persister/entity/AbstractEntityPersister.java

```diff
@@ -4136,60 +4139,13 @@ protected Map getLoaders() {
 	protected void createLoaders() {
 		final Map loaders = getLoaders();
 
-		UniqueEntityLoader readLoader = createEntityLoader( LockMode.READ );
-		loaders.put( LockMode.READ, readLoader );
+		noneLockLoader = createEntityLoader( LockMode.NONE );
+		loaders.put( LockMode.NONE, noneLockLoader );
 
-		loaders.put( LockMode.NONE, createEntityLoader( readLoader, LockMode.NONE ) );
+		readLockLoader = createEntityLoader( noneLockLoader, LockMode.READ );
+		loaders.put( LockMode.READ, readLockLoader );
```

Comment:
It's not necessary to add these 2 loaders to the map but I thought it might be safer considering the map is exposed to the subclasses.